### DISCUSSION TITLE
Determining which diagnosis rooms are most effective

### DISCUSSION THREAD

**Discussion by MandrakeHorse** [CorsixTH/CorsixTH#2417] [General Diagnostics]
The purpose of this discussion is to determine which diagnosis rooms are the most effective at diagnosing patients.

Thanks to @mugmuggy's investigations for https://github.com/CorsixTH/CorsixTH/issues/1303#issuecomment-358797155, it was discovered that the calculation for diagnosis progress includes a "room factor". Each type of diagnosis room can apply a small or large bonus/penalty to progress depending on the patient's illness. Some of these modifiers make logical sense (such as X-rays being ineffective with Invisible patients or psychiatrists not being able to understand patients with Slack Tongue) while others seem more arbitrary. The table of room bonuses is reproduced below and includes a row for the total sum of all these modifiers (including that of the unfinished Alien DNA disease):

<html><body>

Disease | Scanner | Blood | Cardio | Xray | Ultra | General | Ward | Shrink
-- | -- | -- | -- | -- | -- | -- | -- | --
BLOATY_HEAD | 400 | 0 | 0 | 0 | 0 | 0 | 200 | 0
HAIRYITUS | 200 | 0 | 0 | 0 | 0 | 0 | 400 | 0
ELVIS | 0 | 0 | 0 | 0 | 0 | 0 | 200 | 400
INVIS | 0 | 400 | 0 | -200 | 0 | 0 | 0 | 0
RADIATION | -200 | 400 | 0 | -400 | 0 | 0 | 0 | 0
SLACK_TONGUE | 400 | 0 | 200 | 0 | 0 | 0 | 0 | -200
ALIEN | 200 | 400 | 0 | 200 | 200 | 0 | 0 | -400
BROKEN_BONES | 0 | 0 | 200 | 400 | 0 | 0 | 0 | 0
BALDNESS | 0 | 400 | 0 | 0 | 0 | 0 | 200 | -200
DISCRETE_ITCHING | 0 | 400 | 0 | 0 | 0 | 0 | 0 | 0
JELLYITUS | 0 | 0 | 0 | 0 | 0 | 400 | 0 | 0
SLEEPING_ILLNESS | 400 | 0 | 0 | 0 | 0 | 0 | 200 | 0
PREGNANT | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0
TRANSPARENCY | 0 | 0 | 0 | 0 | 0 | 0 | 400 | 0
UNCOMMON_COLD | 0 | 0 | 0 | 0 | 0 | 400 | 0 | 0
BROKEN_WIND | 0 | 0 | 0 | 0 | 400 | 0 | 0 | 0
SAPRE_RIBS | 0 | 0 | 0 | 400 | 0 | 0 | 0 | 0
KIDNEY_BEANS | 0 | 0 | 0 | 200 | 400 | 0 | 0 | 0
BROKEN_HEART | 0 | 0 | 400 | 0 | 200 | 0 | 0 | 0
RUPTURED_NODULES | 400 | 0 | 0 | 0 | 200 | 0 | 0 | 0
MULTIPLE_TV_PERSONALITIES | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 400
INFECTIOUS_LAUGHTER | 0 | 200 | 0 | 0 | 0 | 0 | 0 | 400
CORRUGATED_ANKLES | 0 | 0 | 400 | 0 | 0 | 0 | 0 | 0
CHRONIC_NOSEHAIR | 400 | 0 | 0 | 0 | 0 | 0 | 200 | 0
3RD_DEGREE_SIDEBURNS | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 400
FAKE_BLOOD | 0 | 400 | 0 | 0 | 0 | 0 | 0 | 0
GASTRIC_EJECTIONS | 0 | 0 | 0 | 0 | 0 | 0 | 400 | 0
THE_SQUITS | 0 | 0 | 0 | 0 | 0 | 400 | 200 | 0
IRON_LUNGS | 200 | 0 | 0 | 400 | 0 | 0 | 0 | 0
SWEATY_PALMS | 0 | 0 | -400 | 0 | 0 | 0 | 0 | 400
HEAPED_PILES | 0 | 0 | 0 | 0 | 0 | 0 | 400 | 0
GUT_ROT | 0 | 0 | 0 | 0 | 400 | 0 | 0 | 0
GOLF_STONES | 0 | 0 | 0 | 200 | 400 | 0 | 0 | 0
UNEXPECTED_SWELLING | 400 | 0 | 0 | 0 | 0 | 0 | 0 | 0
**TOTAL**|**2800**|**2600**|**800**|**1200**|**2200**|**1200**|**2800**|**1600**

</body>
</html>

By considering the room factor alone, we can see that the largest bonuses to diagnosis progress are provided by the Scanner and (perhaps somewhat counter-intuitively) the Ward. This is closely followed by the Blood Machine, then Ultra Scanner, Psychiatric Room, then General Diagnosis and X-ray, and then finally Cardio in last place. This generally supports the argument that the more advanced/expensive diagnostic rooms are generally better at diagnosing patients, since these rooms provide the most bonuses to diagnosis progress and have little-to-zero chance of penalty.

**Comment by ARGAMX** [comment]
Are these calculations from the original TH game?
They not applied in CTH?
